Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiablePayload.swift

```diff
@@ -0,0 +1,224 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+protocol IdentifiablePayload {
+    var databaseId: String? { get }
+    static var modelClass: (NSManagedObject & IdentifiableModel).Type? { get }
+    func fillIds(cache: inout [String: Set<String>])
+}
+
+extension IdentifiablePayload {
+    func addId(cache: inout [String: Set<String>]) {
+        guard let databaseId = databaseId, let modelClassName = Self.modelClass?.className else { return }
+        var ids = (cache[modelClassName] ?? Set<String>())
+        ids.insert(databaseId)
+        cache[modelClassName] = ids
+    }
+
+    func recursivelyGetAllIds() -> [String: Set<String>] {
+        var cache: [String: Set<String>] = [:]
+        fillIds(cache: &cache)
+        return cache
+    }
+
+    func getPayloadToModelIdMappings(context: NSManagedObjectContext) -> PreWarmedCache {
+        let payloadIdsMappings = recursivelyGetAllIds()
+
+        var modelToIdMappings: [String: [String: NSManagedObjectID]] = [:]
```

Comment:
```suggestion
        var preWarmedCache: [String: [String: NSManagedObjectID]] = [:]
```
I'm confused by the `modelToIdMappings`, it's more like `modelToIdToModelIdMappings` (which is just more confusing)